tests/Unit/Models/Litter/LitterCategoryTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Tests\Unit\Models\Litter;
+
+
+use App\Models\Photo;
+use Illuminate\Support\Facades\Schema;
+use Tests\TestCase;
+
+class LitterCategoryTest extends TestCase
+{
+    public function getCategories(): array
+    {
+        return array_map(function ($category) {
+            return [$category];
+        }, Photo::categories());
+    }
+
+    /**
+     * @dataProvider getCategories
+     * @param $category
+     */
+    public function test_categories_have_identical_types_as_their_schema($category)
+    {
+        $photo = Photo::factory()->make();
+        $model = $photo->$category()->make([]);
+        $types = $model->types();
```

Comment:
We should use a static method to get types, same as we do for categories

Reply:
In this case we need the `$model` to get the `$model->tableName()` further below. I converted the types() on the categories into static methods, but since we have the categories `$model`, we can still call types() from it.